File: csharp/901.online-stock-span.cs
/*
 * @lc app=leetcode id=901 lang=csharp
 *
 * [901] Online Stock Span
 */

using System.Collections.Generic;

// @lc code=start
public class StockSpanner {

    private Stack<(int Value, int Index)> Stack;
    private int Index;

    public StockSpanner() {
        Stack = new Stack<(int Value, int Index)>();
        Stack.Push((Value: int.MaxValue, Index: -1));
        Index = 0;
    }
    
    public int Next(int price) {
        while (Stack.Peek().Value <= price) {
            Stack.Pop();
        }
        int ret = Index - Stack.Peek().Index;
        Stack.Push((Value: price, Index: Index++));
        return ret;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.Next(price);
 */
// @lc code=end

File: java/901.online-stock-span.java
/*
 * @lc app=leetcode id=901 lang=java
 *
 * [901] Online Stock Span
 */

// @lc code=start
class StockSpanner {

    Stack<int[]> stack;

    public StockSpanner() {
        stack = new Stack<int[]>();
    }
    
    public int next(int price) {
        int ret = 1;
        while (stack.size() > 0 && stack.peek()[0] <= price) {
            ret += stack.pop()[1];
        }
        stack.push(new int[] { price, ret });
        return ret;
    }
}

/**
 * Your StockSpanner object will be instantiated and called as such:
 * StockSpanner obj = new StockSpanner();
 * int param_1 = obj.next(price);
 */
// @lc code=end

